download/downloadSpec.go

```diff
@@ -161,3 +168,36 @@ func detectLinuxId() (string, error) {
 		return "", &UnsupportedSystemError{msg: "invalid linux version '" + id + "'"}
 	}
 }
+
+func readKeyValuePairs(r io.Reader) (content map[string]string, err error) {
```

Comment:
You don't have to return the variable names here, this should suffice:

```suggestion
func readKeyValuePairs(r io.Reader) (map[string]string, error) {
    content := make(map[string]string)
```

Alternatively, if naming the returned variables in the function definition, you can simply `return` instead of `return content, scanner.Err()`